### PR TITLE
[SPARK-49688][CONNECT][TESTS] Fix a sporadic `SparkConnectServiceSuite` failure

### PR DESCRIPTION
**Pull request by ghost**
### What changes were proposed in this pull request?

Add a short wait loop to ensure that the test pre-condition is met. To be specific, VerifyEvents.executeHolder is set asynchronously by MockSparkListener.onOtherEvent whereas the test assumes that VerifyEvents.executeHolder is always available.

### Why are the changes needed?

For smoother development experience.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

SparkConnectServiceSuite.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by changgyoopark-db** [comment]
@jdesjean Hi, nice to meet you! Could you review this trivial change? Thanks!

File: sql/connect/server/src/test/scala/org/apache/spark/sql/connect/planner/SparkConnectServiceSuite.scala [hunk 1]
```diff
@@ -873,8 +873,11 @@ class SparkConnectServiceSuite
     val listenerBus = sparkContext.listenerBus
     val LISTENER_BUS_TIMEOUT = 30000
     def executeHolder: ExecuteHolder = {
-      assert(listener.executeHolder.isDefined)
-      listener.executeHolder.get
+      // An ExecuteHolder will eventually be set
+      Eventually.eventually(timeout(1.seconds)) {
```
Comment (dongjoon-hyun) [inline sql/connect/server/src/test/scala/org/apache/spark/sql/connect/planner/SparkConnectServiceSuite.scala:877]: If we need this, `1.seconds` is enough? Could you provide the failure error message or track trace which you want to fix in this PR?

**Comment by dongjoon-hyun** [comment]
Also, cc @hvanhovell , too.

File: sql/connect/server/src/test/scala/org/apache/spark/sql/connect/planner/SparkConnectServiceSuite.scala (hunk 1 above)
Comment (changgyoopark-db) [reply sql/connect/server/src/test/scala/org/apache/spark/sql/connect/planner/SparkConnectServiceSuite.scala:877]: That makes sense. I'll change this and line 894 to 10 to seconds + add an error message.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by dongjoon-hyun** [comment]
Thank you, @changgyoopark-db and @HyukjinKwon .

**Pull request closed without merging**